### ISSUE TITLE
Blackhole CNN bringup

### ISSUE THREAD

**Issue by abhullar-tt** [tenstorrent/tt-metal#11124]
This issue is meant to act as master list for ops we need to enable CNNs on BH

@mywoodstock helped put together a list Resnet ops/ ops to prioritize for CNNs:
- BinaryDeviceOperation
- InterleavedToSharded
- Matmul
- Move
- PadOnHost
- Reduce
- Reshard
- Tilize
- TilizeWithValPadding
- Untilize
- UntilizeWithUnpadding

CNN specific ops:
- [x] Halo: This is run as part of below two ops (incl. in their unit tests.)
- [x] MaxPool2D: `pytest tests/ttnn/unit_tests/operations/test_maxpool2d.py` -  6 fail; 48 pass
- [x] Conv2D: `pytest tests/ttnn/unit_tests/operations/test_new_conv2d.py` - 12 fail; 212 pass

I believe all these ops should have relevant pytests 

FYI @rtawfik01 @davorchap @jvasilje

**Comment by rtawfik01** [comment]
@mywoodstock is it also possible to add a test to run for each of the above?

**Comment by pgkeller** [comment]
marking P0 as this gates BH checkout

**Comment by pgkeller** [comment]
Assigned Abhinav.  See also these 2 issues for llk work:
 - [] #10498
 - [] #10052

**Comment by rtawfik01** [comment]
Hi I closed #10498, which enables a lot of maxpool/conv tests to run now, but now we need a list of the unit tests for each of the above ops, and update them. I have tried running some different unit tests and some need BH updates: https://github.com/tenstorrent/tt-metal/issues/11684

**Comment by mywoodstock** [comment]
I updated the description with the unit tests for the CNN specific ops (maxpool, conv, halo.)

**Comment by mywoodstock** [comment]
Current status: Blocked on TM ops' uplift to address the alignment issues.
max_pool2d: channels = 16 (all others pass)
conv2d: bfp8_b cases (bfloat16 pass)

**Comment by abhullar-tt** [comment]
@mywoodstock is this a stale issue?

**Comment by mywoodstock** [comment]
@abhullar-tt No, this is where we're tracking the CNN ops instead of on the master list

**Comment by mywoodstock** [comment]
Current status: We have all conv unit tests and maxpool unit tests passing, except for shallow data (c = 16), which give bad pcc due to alignment issues in TMs. Naif has a PR to be merged to fix those. Once thats in, we will be able to try out CNN models.

**Comment by nvelickovicTT** [comment]
@mywoodstock seems this is in good state and haven't had issues for the LLK team. I'll remove the LLK label then, and if a particular LLK issue appears, track it separately.

**Comment by pgkeller** [comment]
@mywoodstock is this resolved?

**Comment by mywoodstock** [comment]
Still ongoing.
Latest statuses:
`conv2d` unit tests: `8 failed, 456 passed`. Only the cases with shallow depth (c = 16) have PCC error.

**Comment by mywoodstock** [comment]
BH support for CNNs is in `main`.
